src/test/java/com/commercetools/emailprocessor/email/EmailProcessorTest.java

```diff
@@ -0,0 +1,232 @@
+        assertEquals(statistic.getProcessedEmails(), 0);
+        assertEquals(statistic.getSuccessfulSendedEmails(), 0);
+        assertEquals(statistic.getTemporarilyErrors(), 0);
+
+    }
+
+    @Test
+    public void shouldNotProcessEmails2() throws Exception {
+        tenantConfiguration.setClient(mockSphereClient(Collections.emptyList()));
+        Statistics statistic = emailProcessor.processEmails(tenantConfiguration).toCompletableFuture().join();
+
+        assertEquals(statistic.getProcessedEmails(), 0);
+        assertEquals(statistic.getSuccessfulSendedEmails(), 0);
+        assertEquals(statistic.getTemporarilyErrors(), 0);
+
+    }
+
+    @Test
+    public void shouldcallApiEndpoint() throws Exception {
+        Mockito.doCallRealMethod().when(emailProcessor)
+            .callApiEndpoint(Mockito.anyString(), Mockito.any(TenantConfiguration.class));
+        final int httpStatus = 200;
+        final String id = "123";
+        final String tenantid = "testTenant";
+        final String url = "http://www.anyurl.de";
+        MockUrlStreamHandler handler = new MockUrlStreamHandler(url);
+        URL.setURLStreamHandlerFactory(handler);
+        MockHttpUrlConnection httpUrlConnection = handler.getConnection();
+        TenantConfiguration configuration = new TenantConfiguration();
+        configuration.setHttpUrlConnection(httpUrlConnection);
+        configuration.setProjectKey(tenantid);
+        int result = emailProcessor.callApiEndpoint(id, configuration);
+        assertEquals(httpStatus, result);
+        assertEquals(url, httpUrlConnection.getURL().toString());
+        assertEquals(IOUtils.toString(httpUrlConnection.getInputStream(), Charset.defaultCharset()),
+            "emailid=" + id + "&tenantid=" + tenantid);
+
+    }
+
+    @SuppressWarnings("unchecked")
+    private SphereClient mockSphereClient(final List<CustomObject<JsonNode>> customObjects) {
+        SphereClient client = mock(SphereClient.class);
+        final PagedQueryResult<CustomObject<JsonNode>> queryResult = PagedQueryResult.of(customObjects);
+        when(client.execute(isA(CustomObjectQuery.class))).thenReturn(CompletableFuture.completedFuture(queryResult));
+        return client;
+    }
+
+    @SuppressWarnings("unchecked")
+    private CustomObject<JsonNode> createCustomObject(final String customobjectid, final String status, final int
+        endPointtatus)
+        throws Exception {
+        when(emailProcessor.callApiEndpoint(customobjectid, tenantConfiguration)).thenReturn(endPointtatus);
+        JsonNode jsonNode = SphereJsonUtils.parse(String.format("{\"status\":\"%s\"}", status));
+        CustomObject<JsonNode> customObject = mock(CustomObject.class);
+        when(customObject.getId()).thenReturn(customobjectid);
+        when(customObject.getValue()).thenReturn(jsonNode);
+        return customObject;
+    }
+
+    public static class MockUrlStreamHandler extends URLStreamHandler implements URLStreamHandlerFactory {
```

Comment:
Does it make sense to move this to it's separate class?

Reply:
no

Reply:
no,because this class isn`t used anywhere else.